### PR TITLE
DOC: Add example with ``numpy_nullable`` to ``pd.read_xml()``

### PR DESCRIPTION
**Pull request by linus-md**
- [x] [closes](https://github.com/noatamir/pyladies-berlin-sprints/issues/4)

### CODE REVIEW

File: doc/source/user_guide/io.rst [hunk 1]
```diff
@@ -3279,6 +3279,32 @@ of reading in Wikipedia's very large (12 GB+) latest article data dump.
 
     [3578765 rows x 3 columns]
 
+The following is an example with ``use_nullable_dtypes=True``:
```
Comment (mroeschke) [inline doc/source/user_guide/io.rst:3282]: Could you move these to the docstring instead of this user guide? (in addition to your other PRs)

**Review by mroeschke — CHANGES_REQUESTED** [review]

File: doc/source/user_guide/io.rst (hunk 1 above)
Comment (linus-md) [reply doc/source/user_guide/io.rst:3282]: Will do.

Comment (linus-md) [reply doc/source/user_guide/io.rst]: @mroeschke I moved the xml and json example but in the ``read_html`` example section it says ```See the :ref:`read_html documentation in the IO section of the docs
    <io.read_html>` for some examples of reading in HTML tables.```

Comment (mroeschke) [reply doc/source/user_guide/io.rst]: It would be good to move those too over to the docstring

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @linus-md

**Pull request merged**